component/core/src/main/java/org/exoplatform/social/core/binding/impl/GroupSpaceBindingServiceImpl.java

```diff
@@ -119,47 +118,36 @@ public GroupSpaceBindingQueue findFirstGroupSpaceBindingQueue() {
    */
   @Override
   public List<UserSpaceBinding> findUserBindingsByGroup(String group, String userName) {
-    LOG.info("Retrieving user bindings for user : " + userName + " with group : " + group);
+    LOG.debug("Retrieving user bindings for user : " + userName + " with group : " + group);
     return groupSpaceBindingStorage.findUserSpaceBindingsByGroup(group, userName);
   }
 
   /**
    * {@inheritDoc}
+   * @return
    */
   @Override
-  public void saveGroupSpaceBindings(List<GroupSpaceBinding> groupSpaceBindings) {
-    LOG.info("Saving space bindings ("+groupSpaceBindings.size()+" to save)");
+  public GroupSpaceBinding saveGroupSpaceBinding(GroupSpaceBinding groupSpaceBinding) {
+    LOG.debug("Saving group space binding between spaceId: {} and group: {}.", groupSpaceBinding.getSpaceId());
+    GroupSpaceBinding binding;
     try {
-      for (GroupSpaceBinding groupSpaceBinding : groupSpaceBindings) {
-        groupSpaceBinding = groupSpaceBindingStorage.saveGroupBinding(groupSpaceBinding, groupSpaceBinding.getId() == -1);
-        ListAccess<User> groupMembersAccess =
-                                            organizationService.getUserHandler().findUsersByGroupId(groupSpaceBinding.getGroup());
-        User[] users = groupMembersAccess.load(0, groupMembersAccess.getSize());
-        for (User user : users) {
-          // add user binding for this space
-          List<UserSpaceBinding> userSpaceBindings = new LinkedList<>();
-          UserSpaceBinding userSpaceBinding = new UserSpaceBinding();
-          userSpaceBinding.setGroupBinding(groupSpaceBinding);
-          userSpaceBinding.setUser(user.getUserName());
-          userSpaceBindings.add(userSpaceBinding);
-          this.saveUserBindings(userSpaceBindings);
-        }
-      }
+      binding = groupSpaceBindingStorage.saveGroupSpaceBinding(groupSpaceBinding);
     } catch (Exception e) {
       LOG.error("Error Binding" + e);
       throw new RuntimeException("Failed saving groupSpaceBinding", e);
     }
+    return binding;
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
   public void saveUserBindings(List<UserSpaceBinding> userSpaceBindings) {
-    LOG.info("Saving user bindings, ("+userSpaceBindings.size()+" to save)");
+    LOG.debug("Saving user bindings, ({}) to save", userSpaceBindings.size());
     try {
       for (UserSpaceBinding userSpaceBinding : userSpaceBindings) {
-        this.bindUserToSpace(userSpaceBinding);
+        bindUserToSpace(userSpaceBinding);
```

Comment:
```suggestion
        groupSpaceBindingStorage.saveUserBinding(userSpaceBinding);
```
No need to create a new function here. just call the storage part

Reply:
Okay